### ISSUE TITLE
`unformatOnSubmit` does not unformat fields located outside of form DOM subtree

### ISSUE THREAD

**Issue by gap777** [autoNumeric/autoNumeric#768]
### Current behavior
Autonumeric won't recognize input elements in the DOM, but outside the form subtree, as a part of the form.

### Expected behavior
Autonumeric would find the `parentForm` using the `element.form` instead of searching the DOM `parentNode`

### Steps to reproduce the problem
1. Use autoNumeric version `4.8.1`
2. In the browser `Chrome` version `Version 111.0.5563.146 (Official Build) (arm64)`
3. On the `MacOS` operating system, version `13.3.1 (a)`
4. Instantiate autonumeric on a field OUTSIDE of the form subtree, but connected to the form via the `form` attribute. Submit the form and see that the field is not unformatted.

### Link to live example (ie. [Codepen](http://codepen.io/pen/))
[<Providing a live example of your problem that reproduce what you are experiencing usually means we'll be able to fix your issue *much more quickly* than if you don't.>](https://codepen.io/gap777/pen/poxLrax?editors=1011)

### Proposed solution

We are monkey-patching to work-around this, but I don't understand why AN doesn't do this out of the gate:

```js
AutoNumeric.prototype._getParentForm = function() {
      return this.domElement.form
    }
    AutoNumeric.prototype._getFormAutoNumericChildren = function(formElement) {
      const inputList = [... formElement.elements ];

      return inputList.filter(input => this.constructor.isManagedByAutoNumeric(input));
    }
```

**Comment by gap777** [comment]
I can open a PR with my proposed solution, if you agree with the approach. Let me know.

**Comment by AlexandreBonneau** [comment]
Well, I did not know about `Element.form` since the official MDN documentation [does not mention it](https://developer.mozilla.org/en-US/docs/Web/API/Element), my bad ;x

This could simplify the `_getParentForm()` function a lot, however it seems it breaks some of the unit tests.
I've adapted your solution to keep all the tests green so far, but haven't yet had time to check the end-to-end tests.

Since I'm editing this part of the code, I'm updating it so that the `contenteditable` nodes are correctly selected as well, even if outside the form element and despite the fact that the 'form' attribute is seldom permitted on those.

On a side note, I've updated your [example](https://codepen.io/AnotherLinuxUser/pen/PoyaoVY?editors=1010) with the usual calls to create AutoNumeric elements, as well as some `contenteditable` elements.
